### PR TITLE
Adding header to each colour of line to indicate the divsion of codechef contest

### PR DESCRIPTION
**Pull request by 25lahari**
Issue - https://github.com/chinesedfan/clist-ex/issues/7
Files Changes:
src\pages\ProblemPage.tsx

### CODE REVIEW

File: src/pages/ProblemPage.tsx [hunk 1]
```diff
@@ -25,17 +25,62 @@ const items: TabsProps['items'] = [
...
         label: 'CodeChef',
         children: (
-            <ProblemFilter radios={[
-                { label: 'All', value: '' },
-                { label: 'Rated for All', value: 'All' },
-                { label: 'Rated till 6-stars', value: '6.[Ss]tars' },
-                { label: 'Rated till 5-stars', value: '5.[Ss]tars' },
-                { label: 'Rated for Divs 2, 3 & 4', value: 'Divs 2' },
-            ]}></ProblemFilter>        
+            <div style={{ position: 'relative', padding: '20px' }}>
```
Comment (chinesedfan) [inline src/pages/ProblemPage.tsx]: Only changing the top and bottom padding will be better.

File: src/pages/ProblemPage.tsx [hunk 2]
```diff
@@ -25,17 +25,62 @@ const items: TabsProps['items'] = [
...
+                        top: '-12px',
+                        right: '15px',
+                        padding: '8px',
+                        borderRadius: '4px', 
+                        width: '300px',
+                        display : "flex",
+                        flexWrap : "wrap",
+                        overflowX: 'auto', // Adds horizontal scroll if needed
+                        //whiteSpace: 'nowrap', // Ensures content stays in a single line if necessary
+                        //backgroundColor : 'red',
```
Comment (chinesedfan) [inline src/pages/ProblemPage.tsx]: Remove useless styles and keep same single quotes.

File: src/pages/ProblemPage.tsx [hunk 3]
```diff
@@ -25,17 +25,62 @@ const items: TabsProps['items'] = [
...
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                </div>
+            </div>
         ),
     },
 ];
-
```
Comment (chinesedfan) [inline src/pages/ProblemPage.tsx]: Don't change if not necessary, as well as the deleted empty line in the end of file.

File: src/pages/ProblemPage.tsx [hunk 4]
```diff
@@ -25,17 +25,62 @@ const items: TabsProps['items'] = [
...
+                }}>
+                    <table style={{ width: '100%', borderCollapse: 'collapse' ,tableLayout: 'fixed'}}>
+                        <tbody>
+                            <tr>
+                                <td style={{ padding: '0 5px', textAlign: 'center' }}>
+                                    <div style={{ display: 'flex', alignItems: 'center' }}>
+                                        <span style={{ width: '20px', height: '5px', backgroundColor: 'red', marginRight: '5px' }}></span>
+                                        <span >Div 1</span>
+                                    </div>
+                                </td>
```
Comment (chinesedfan) [inline src/pages/ProblemPage.tsx]: How about using a loop like ProblemList.jsx? And use class names to control different styles by SCSS variables.

File: src/pages/ProblemPage.tsx [hunk 5]
```diff
@@ -25,17 +25,62 @@ const items: TabsProps['items'] = [
...
-            ]}></ProblemFilter>        
+            <div style={{ position: 'relative', padding: '20px' }}>
+                <ProblemFilter radios={[
+                    { label: 'All', value: '' },
+                    { label: 'Rated for All', value: 'All' },
+                    { label: 'Rated till 6-stars', value: '6.[Ss]tars' },
+                    { label: 'Rated till 5-stars', value: '5.[Ss]tars' },
+                    { label: 'Rated for Divs 2, 3 & 4', value: 'Divs 2' },
+                ]}></ProblemFilter>
+                <div style={{
```
Comment (chinesedfan) [inline src/pages/ProblemPage.tsx]: Maybe extract added elements as a new component in src/components.

**Review by chinesedfan — CHANGES_REQUESTED** [review]
Very sweet feature! Left some comments, and you can exclude changes in yarn.lock or package-lock.json.

File: src/components/LegendComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+const data = [
+    { color: 'red', label: 'Div 1' },
+    { color: 'orange', label: 'Div 2' },
+    { color: '#40826D', label: 'Div 3' },
+    { color: 'grey', label: 'Div 4' },
+  ];
+  
+  const LegendComponent = () => (
```
Comment (chinesedfan) [inline src/components/LegendComponent.tsx]: Add React.memo

File: src/components/LegendComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+                </div>
+              </td>
+            ))}
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  );
+  
+  export default LegendComponent;
```
Comment (chinesedfan) [inline src/components/LegendComponent.tsx]: export with name.

File: src/components/LegendComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
+const data = [
+    { color: 'red', label: 'Div 1' },
+    { color: 'orange', label: 'Div 2' },
+    { color: '#40826D', label: 'Div 3' },
+    { color: 'grey', label: 'Div 4' },
+  ];
```
Comment (chinesedfan) [inline src/components/LegendComponent.tsx]: Keep 4 spaces as indentation.

File: src/styles/problem.scss [hunk 9]
```diff
@@ -111,3 +111,34 @@ td.problem-cell {
 .problem-div-4 {
```
Comment (chinesedfan) [inline src/styles/problem.scss:111]: I'd like something like this. Different class names with SCSS variables.

**Review by chinesedfan — CHANGES_REQUESTED** [review]
Thanks for your patient updates. Would you mind update again and fix those code style changes?

File: src/styles/problem.scss (hunk 9 above)
Comment (chinesedfan) [reply src/styles/problem.scss:111]: @25lahari Remind. There are still some conversations need to be resolved.

Comment (25lahari) [reply src/styles/problem.scss:111]: I sorry but not able to understand what u mean can you elaborate please.

Comment (chinesedfan) [reply src/styles/problem.scss:111]: I mean I have defined some color variables in the beginning of this file. So you can reuse them in your css classes, and update the JSX like following,

```
<div className={'problem-div-' + x }
```

File: src/styles/problem.scss [hunk 10]
```diff
@@ -111,3 +111,34 @@ td.problem-cell {
...
     background: grey;
 }
+.div-heading-container {
+    position: absolute;
+    top: -12px;
+    right: 15px;
+    padding: 8px;
+    border-radius: 4px;
+    width: 300px;
+  }
```
Comment (chinesedfan) [inline src/styles/problem.scss]: Add spaces here is not correct.

Comment (chinesedfan) [reply src/styles/problem.scss]: Let's check all unfolded comments, and click "Resolve conversation" if you think it has been fixed.

File: src/styles/problem.scss (hunk 9 above)
Comment (25lahari) [reply src/styles/problem.scss:111]: I have updated the changes.

**Pull request merged**